react/features/base/settings/reducer.js

```diff
@@ -116,17 +116,24 @@ function _initSettings(featureState) {
     }, settings);
 
     if (!browser.isReactNative()) {
+        const persistedSettings = JSON.parse(
+            window.localStorage.getItem('features/base/settings') || '{}');
+
         // Browser only
-        const localFlipX
-            = JSON.parse(window.localStorage.getItem('localFlipX') || 'true');
-        const cameraDeviceId
-            = window.localStorage.getItem('cameraDeviceId') || '';
-        const micDeviceId = window.localStorage.getItem('micDeviceId') || '';
+        const localFlipX = persistedSettings.localFlipX
```

Comment:
This is a boolean and if persistedSettings.localFilpX is false, it can be overwritten by the legacy value. Not sure if this flaw was in the original implementation too, but just pointed it out now.

Reply:
Good point. I'll have to fix that.